src/openms_gui/source/VISUAL/SpectraTreeTab.cpp

```diff
@@ -182,19 +182,28 @@ namespace OpenMS
     }
   }
 
+  void SpectraTreeTab::updateIndexFromCurrentLayer()
+  {
+    std::cout << "Spectra tab view activated to go to position " << layer_->getCurrentSpectrumIndex() << std::endl;
+    spectra_treewidget_->setTreePosition(layer_->getCurrentSpectrumIndex());
+  }
+
   void SpectraTreeTab::itemSelectionChange_(QTreeWidgetItem* current, QTreeWidgetItem* previous)
   {
     /*	test for previous == 0 is important - without it,
         the wrong spectrum will be selected after finishing
         the execution of a TOPP tool on the whole data */
-    if (current == nullptr || previous == nullptr)
+    // WUT????
+    std::cout << "itemselection in specra tabview changed:" << std::endl;
+    if (current == nullptr)
```

Comment:
yes because it is too strict. why shouldn't we be allowed to change the selection if nothing was selected yet. Previous is not even used..
The person should fix it in the TOPP tool calling code if this really is an issue.

Reply:
I vaguely recall that this was needed at some point (e.g., there were some issues when switching between tabs and datasets). Maybe keep an eye on it if this causes strange behavior